src/model/element-base.ts

```diff
@@ -95,7 +131,180 @@ export abstract class ElementBase implements Feature {
    * single list. A mixin can be applied more than once, each time its
    * members override those before it in the prototype chain.
    */
-  mixins: Reference[] = [];
+  readonly mixins: ImmutableArray<Reference>;
+
+  constructor(init: ElementBaseInit, document: Document) {
+    const {
```

Comment:
note, you can put this in the parameter list, but the way you have it now may be more legible